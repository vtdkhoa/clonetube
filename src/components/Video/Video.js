import React from 'react'
import './Video.scss'

const BASE_EMBED_URL = 'https://www.youtube.com/embed/'

const Video = props => {
  if (!props.id) return null

  const embedUrl = `${BASE_EMBED_URL}${props.id}?autoplay=1`

  return (
    <div className="video-container">
      <div className="video">
      <iframe
        className="video-player"
        title={`${props.id}`}
        width={'560'}
        height={'315'}
        src={embedUrl}
        frameBorder='0'
        allow='autoplay; encrypted-media'
        allowFullScreen
      />
      </div>
    </div>
  )
}

export default Video