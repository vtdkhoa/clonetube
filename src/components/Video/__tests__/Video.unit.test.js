import React from 'react'
import Video from '../Video'
import { shallow } from 'enzyme'

describe('Video', () => {
  test('renders video component correctly', () => {
    const wrapper = shallow(
      <Video id="12345"/>
    )
    expect(wrapper).toMatchSnapshot()
  })

  test('renders null if id prop not specified', () => {
    const wrapper = shallow(<Video/>)
    expect(wrapper).toMatchSnapshot()
  })
})