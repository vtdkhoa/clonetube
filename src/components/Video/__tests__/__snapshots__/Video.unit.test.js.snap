// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Video renders null if id prop not specified 1`] = `""`;

exports[`Video renders video component correctly 1`] = `
<div
  className="video-container"
>
  <div
    className="video"
  >
    <iframe
      allow="autoplay; encrypted-media"
      allowFullScreen={true}
      className="video-player"
      frameBorder="0"
      height="315"
      src="https://www.youtube.com/embed/12345?autoplay=1"
      title="12345"
      width="560"
    />
  </div>
</div>
`;
